interfaces/repo.go

```diff
@@ -516,3 +516,126 @@ func (r *Repository) securitySnippetsForSnap(snapName string, securitySystem Sec
 	}
 	return snippets, nil
 }
+
+type addSnapError struct {
```

Comment:
Plugs and slots must not have the same name. This is an invariant we defined very early on.. it's even in the spec! :)